platform/android/export/export_plugin.cpp

```diff
@@ -3150,10 +3236,125 @@ Error EditorExportPlatformAndroid::export_project_helper(const Ref<EditorExportP
+					pad = _get_pad(4, string_len);
+					len += (4 + string_len + pad);
+				}
+				len += 64; // Encryption overhead (hash + iv + size).
+				if (len % 16) {
+					len += 16 - (len % 16); // Alignment.
+				}
+				dir_data.resize(len);
+				memset(dir_data.ptrw(), 0, len);
+
+				Ref<FileAccessMemory> fmem;
+				fmem.instantiate();
+				ERR_FAIL_COND_V(fmem.is_null(), ERR_CANT_CREATE);
+				err = fmem->open_custom(dir_data.ptrw(), dir_data.size());
+				ERR_FAIL_COND_V(err != OK, ERR_CANT_CREATE);
+
+				fmem->store_32(DIR_HEADER_MAGIC);
+				fmem->store_32(PACK_FORMAT_VERSION);
+				fmem->store_32(VERSION_MAJOR);
+				fmem->store_32(VERSION_MINOR);
+				fmem->store_32(VERSION_PATCH);
+				fmem->store_32(PACK_DIR_ENCRYPTED); // flags
+
+				for (int i = 0; i < 16; i++) {
+					//reserved
+					fmem->store_32(0);
+				}
+
+				fmem->store_32(ed.directory.size()); //amount of files
+
+				Ref<FileAccessEncrypted> fae;
+				Ref<FileAccess> fhead = fmem;
+				String script_key = _get_script_encryption_key(p_preset);
+				Vector<uint8_t> key;
+				key.resize(32);
+				if (script_key.length() == 64) {
+					for (int i = 0; i < 32; i++) {
+						int v = 0;
+						if (i * 2 < script_key.length()) {
+							char32_t ct = script_key[i * 2];
+							if (is_digit(ct)) {
+								ct = ct - '0';
+							} else if (ct >= 'a' && ct <= 'f') {
+								ct = 10 + ct - 'a';
+							}
+							v |= ct << 4;
+						}
+
+						if (i * 2 + 1 < script_key.length()) {
+							char32_t ct = script_key[i * 2 + 1];
+							if (is_digit(ct)) {
+								ct = ct - '0';
+							} else if (ct >= 'a' && ct <= 'f') {
+								ct = 10 + ct - 'a';
+							}
+							v |= ct;
+						}
+						key.write[i] = v;
+					}
+				}
+				fae.instantiate();
+				if (fae.is_null()) {
+					add_message(EXPORT_MESSAGE_ERROR, TTR("Save APK"), TTR("Can't create encrypted file."));
+					return ERR_CANT_CREATE;
+				}
+
+				err = fae->open_and_parse(fmem, key, FileAccessEncrypted::MODE_WRITE_AES256, false);
+				if (err != OK) {
+					add_message(EXPORT_MESSAGE_ERROR, TTR("Save APK"), TTR("Can't open encrypted file to write."));
+					return ERR_CANT_CREATE;
+				}
+				fhead = fae;
+
+				for (const KeyValue<String, String> &E : ed.directory) {
+					uint32_t string_len = E.key.utf8().length();
+					uint32_t pad = _get_pad(4, string_len);
+
+					fhead->store_32(string_len + pad);
+					fhead->store_buffer((const uint8_t *)E.key.utf8().get_data(), string_len);
+					for (uint32_t j = 0; j < pad; j++) {
+						fhead->store_8(0);
+					}
+
+					string_len = E.value.utf8().length();
+					pad = _get_pad(4, string_len);
+
+					fhead->store_32(string_len + pad);
+					fhead->store_buffer((const uint8_t *)E.value.utf8().get_data(), string_len);
+					for (uint32_t j = 0; j < pad; j++) {
+						fhead->store_8(0);
+					}
+				}
+
+				if (fae.is_valid()) {
+					fhead.unref();
+					fae.unref();
+				}
+
+				fmem.unref();
+
+				store_in_apk(&ed, "assets/.encrypted/directory", dir_data, 0);
```

Comment:
Hidden directories are usually ignored in the build process. We override this behavior in [our gradle configuration](https://github.com/godotengine/godot/blob/master/platform/android/java/app/build.gradle#LL92-L95), but wherever possible we still want to avoid explicitly using hidden directory if possible.

See https://github.com/godotengine/godot/blob/master/doc/classes/ProjectSettings.xml#L294

Reply:
Changed to `"assets/encrypted/"`.